### ISSUE TITLE
Bug in quantizers/softvq line102

### ISSUE THREAD

**Issue by ezacngm** [Hhhhhhao/continuous_tokenizer#4]
z_q = z_q.view(batch_size, -1, self.e_dim)  #seems problematic
        
        # Reshape back to match original input shape
        if len(z.shape) == 4:
            z_q = torch.einsum('b h w c -> b c h w', z_q)
        
        # Calculate average probabilities
        avg_probs = torch.mean(torch.mean(probs, dim=-1))
        max_probs = torch.mean(torch.max(probs, dim=-1)[0])
       
Hi， 
    Great works！Thanks for sharing your work, but I got one minor bug to report. Here z_q is a 3D tensor after the view method, how could the einsum be applied before reshaping z_q into 4D tensor?

Plz forgive me if this is one stupid question.     

Best,

**Comment by ezacngm** [comment]
Okk so z.shape is not 4, nevermind, sorry
